### ISSUE TITLE
"lightning-grid" requires an older version of "packaging" dependency

### ISSUE THREAD

**Issue by marpaia** [gridai/gridai#92]
## 🐛 Bug

The `lightning-grid` CLI pins a fair few dependencies to an old version. One that is giving me particular trouble is `packaging`. See [Current Behavior](#current-behavior) for more details.

### Expected behavior

I expect to be able to install `lightning-grid` into my local Python environment alongside common libraries like HuggingFace's `transformers` and `datasets` libraries.

### Current behavior

```
ERROR: ResolutionImpossible: for help visit https://pip.pypa.io/en/latest/user_guide/#fixing-conflicting-dependencies
The conflict is caused by:
    The user requested packaging>=20.7
    datasets 1.10.2 depends on packaging
    huggingface-hub 0.0.12 depends on packaging>=20.9
    lightning-grid 0.3.79 depends on packaging==20.7
To fix this you could try to:
1. loosen the range of package versions you've specified
2. remove package versions to allow pip attempt to solve the dependency conflict
```

### To Reproduce

```bash
# install Python 3.9.6
pyenv install 3.9.6

# create and activate a new 3.9.6 virtual environment 
pyenv virtualenv 3.9.6 grid-test-3.9.6
pyenv activate grid-test-3.9.6

# try installing conflicting libraries
pip install datasets lightning-grid
```

This last command will produce a crazy set of conflicts and dependency resolution induced downloads due to lightning-grid pinning a potentially unnecessarily old set of older dependencies.

### Environment

 - OS (e.g., Windows): macOS
 - Browser (e.g., Chrome, Safari): N/A
 - CLI version: Latest (0.3.79)

### Additional context

None to not at the moment but happy to provide more info if necessary.

**Comment by sunitaprakash** [comment]
@marpaia Thanks for reporting the issue. We will improve the dependency requirements.

**Comment by marpaia** [comment]
@sunitaprakash is there an update on this? This is causing another issue in my internal infrastructure again now. I’m afraid that, if y’all don’t manage to use updated versions of dependencies, I’m going to have to remove `grid` from my stack and explore other options. I really like the tool though and I’d love to keep using it. Please help! It should be pretty straight forward to update the dependencies and push a new release.

```
The conflict is caused by:
    The user requested google-resumable-media==1.3.3
    google-cloud-storage 1.41.1 depends on google-resumable-media<3.0dev and >=1.3.0; python_version >= "3.6"
    lightning-grid 0.3.79 depends on google-resumable-media==0.5.1
```

**Comment by luiscape** [comment]
@marpaia we've just fixed this issue on staging. It'll be come out in our next release.

**Comment by luiscape** [comment]
@marpaia here's a [pre-release wheel](https://grid-packages.storage.googleapis.com/grid-cli/21.08.05-27a3d6f42.24120/lightning_grid-21.08.05_27a3d6f42.24120-py2.py3-none-any.whl) in case that helps you get unblocked in between now and when the release comes out. You can install it with:

```
pip install https://grid-packages.storage.googleapis.com/grid-cli/21.08.05-27a3d6f42.24120/lightning_grid-21.08.05_27a3d6f42.24120-py2.py3-none-any.whl
```
We added a test in Python 3.9 and `pip` version 21.2.2 (latest) to make sure it works with your example above. 

We have plans to trim down the dependency requirements of the CLI in the coming month. There's a lot there that can be simplified to prevent these kind of errors. 

I'll close this ticket whenever the release is out.

**Comment by luiscape** [comment]
@marpaia the latest CLI version (`0.3.82`) includes a fix for this issue.

**Comment by marpaia** [comment]
@luiscape this isn't fixed for me unfortunately because `lightning-grid` is still pinning `google-cloud-storage` and other libraries:

```
ERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts.
lightning-grid 0.3.82 requires google-auth==1.22.0, but you have google-auth 1.34.0 which is incompatible.
lightning-grid 0.3.82 requires google-cloud-core==1.3.0, but you have google-cloud-core 1.7.2 which is incompatible.
lightning-grid 0.3.82 requires google-cloud-storage==1.28.1, but you have google-cloud-storage 1.41.1 which is incompatible.
lightning-grid 0.3.82 requires google-resumable-media==0.5.1, but you have google-resumable-media 1.3.3 which is incompatible.
```

Is this necessary? Is there a way that I can upload things to a datastore without having to install any Grid software?

**Comment by rlizzo** [comment]
This has been addressed in our staged release.

**Comment by marpaia** [comment]
Amazing, that sounds awesome! I've backed out usage of grid from our production data pipelines while this is in flight but I'm definitely keen to work it back in so that I can re-enable the automatic upload of dbt-created training data to grid datastores when possible so definitely keep me posted 🙏

**Comment by marpaia** [comment]
Personally, I also have a workaround to build the `grid` CLI into my Airflow base image via `pipx` but I'm in the middle of an Airflow migration right now and that will require all of the dependent dags to be moved to Airflow before I can move the grid upload job there so that has a few dependencies on my end. That said, I'm sure someday there will be useful programatic SDK capabilities for uploading data, running jobs, and downloading artifacts so I think it's still useful and valuable to work out how to make `lightning-grid` as friendly to install as possible. As such, I definitely appreciate the continued effort here.

**Comment by sunitaprakash** [comment]
@marpaia We are working on making the experience better.. Will keep you posted.

**Comment by sunitaprakash** [comment]
Hello @marpaia 
Just wanted to update that our team is working on a fix. We will have a fix for this in the next release. Thanks for your patience.
